include/EASTL/sort.h

```diff
@@ -797,10 +818,10 @@ namespace eastl
 			if(compare(*i, *first))
 			{
 				EASTL_VALIDATE_COMPARE(!compare(*first, *i)); // Validate that the compare function is sane.
-				const value_type temp(*i);
-				*i = *first;
+				value_type temp(eastl::forward<value_type>(*i));
```

Comment:
This one doesn't fail unit tests... but that's an indicator that it needs a unit test... see below.